docs/adr/environments-and-ci.md

```diff
@@ -0,0 +1,78 @@
+# ADR 35580f0d-f429-412b-acef-83655e3cab11: Runtime/engine/host/environment support and CI
+
+## Status
+
+Proposed
+
+## Submitters
+
+- @ctcpip
+
+## Decision Owners
+
+- @expressjs/express-tc
+
+## Context
+
+Express and its libraries were specifically designed to run with Node.js (V8). While some of our libraries can run in other environments (e.g. runtimes, engines, browsers), they are not necessarily supported in all environments. Consequently, our CI systems do not include other environments as part of their testing workflows.
```

Comment:
To disambiguate a bit from "version" and hopefully make it more clear why that is relevant in this ADR:

```suggestion
Express and its libraries were specifically designed to run with Node.js (and the V8 JavaScript Engine). While some of our libraries can run in other environments (e.g. runtimes, engines, browsers), they are not necessarily supported in all environments. Consequently, our CI systems do not include other environments as part of their testing workflows.
```